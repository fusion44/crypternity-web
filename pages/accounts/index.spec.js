import { shallow } from "@vue/test-utils"
import index from "./index"

describe("Accounts Index", () => {
  it("renders the correct markup empty", () => {
    let wrapper = shallow(index)

    let html = wrapper.html()
    expect(html).toMatchSnapshot()
  })

  it("renders the correct markup with 3 accounts", () => {
    let data = {
      propsData: {
        items: [
          {
            id: "1",
            creationDate: "2018-01-16T18:37:17.075765+00:00",
            name: "t1",
            serviceType: "kraken"
          },
          {
            id: "2",
            creationDate: "2018-01-15T19:46:53.293376+00:00",
            name: "t2",
            serviceType: "bitfinex"
          },
          {
            id: "3",
            creationDate: "2018-01-15T19:43:54.055119+00:00",
            name: "t3",
            serviceType: "bitfinex"
          }
        ]
      }
    }

    let wrapper = shallow(index, data)

    let html = wrapper.html()
    expect(html).toMatchSnapshot()
  })
})
