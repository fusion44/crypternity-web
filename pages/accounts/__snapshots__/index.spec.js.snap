// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accounts Index renders the correct markup empty 1`] = `
<div>
  <v-layout column="column" justify-center="justify-center" align-center="align-center">
    <v-flex sm8="sm8" md6="md6">
      <v-data-table headers="[object Object],[object Object],[object Object],[object Object]" items="" class="elevation-1"></v-data-table>
      <v-btn router="router" to="/accounts/add" color="primary">ADD</v-btn>
    </v-flex>
  </v-layout>
</div>
`;

exports[`Accounts Index renders the correct markup with 3 accounts 1`] = `
<div>
  <v-layout column="column" justify-center="justify-center" align-center="align-center">
    <v-flex sm8="sm8" md6="md6">
      <v-data-table headers="[object Object],[object Object],[object Object],[object Object]" items="" class="elevation-1"></v-data-table>
      <v-btn router="router" to="/accounts/add" color="primary">ADD</v-btn>
    </v-flex>
  </v-layout>
</div>
`;
